### ISSUE TITLE
Updates to reject a declaration model 

### ISSUE THREAD

**Issue by jpye-finch** [opencrvs/opencrvs-core#4286]
### Description
After assessing the different reasons. It was deemed no meaningful value is added by capturing if the rejection was a spelling mistake or missing document. If a spelling mistake then the registrar should be updating. So we need another way to assess data quality. Documents will be configured in future to be mandatory/non mandatory in the future

- removes reason for rejection checkboxes
- Adds mark as duplicate checkbox. (If true record can only be archived)
- Add Archive button
- ticket linked to Comment section enhancement
- ticket linked to bug with capturing multiple reasons in record Audit. 
- We wish to rename record audit "Rejected" row to "Sent for updates"
- The archive row in record audit, when clicked for a regular archive should show no comment
- The archive row in record audit, when clicked for a rejection should show the comment in the modal
- The archive row in record audit, when clicked for a rejection where marked as duplicate, should show the comment and pill in the modal

### Acceptance criteria

GIVEN I am a National Reg, Registrar or Reg Agent
WHEN I select Reject
THEN show the reject model

GIVEN the select model is open
WHEN I know comment is added
[THEN disabled the Send for updates and Archive buttons
](https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=9784%3A71401&t=yqYqZJFcLGb566gc-1)

GIVEN the select 'Mark as duplicate'
[THEN disabled the Send for updates buttons
](https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=9784%3A71451&t=yqYqZJFcLGb566gc-1)

GIVEN a record is sent for updates
WHEN in the history view of record audit where we currently see "Rejected", 
THEN we should see "Sent for updates"

GIVEN I click on the "Sent for updates" link in Record Audit
WHEN the model Sent for updates is open
THEN show the comment added

GIVEN a record is archived
WHEN in the model Archived is open in record audit/user audit.
[THEN show the comment added](https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=9784%3A71726&t=icdVRamxXvBzSybW-1)

GIVEN a record is marked as a duplicate and archived
WHEN in the model Archived is open in record audit/user audit.
[THEN show the comment added and < Pill / > Marked as a duplicate 
](https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=9784%3A71672&t=yqYqZJFcLGb566gc-1)
### Design
https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=9561%3A55844&t=yqYqZJFcLGb566gc-1

### Dev tasks

- [ ] Remove the rejectForm reducer as we don't really need to use it. Convert the `RejectRegistrationForm` component to functional one.
- [x] From `reject-registration.ts` file where the form definition is, change the order according to the new design and remove the other options(misspelling, missing_supporting_doc, other) from the CHECKBOX_GROUP. If duplicate checkbox is selected, disable the Send for updates button. - 2
- [x] Add the `Archive` button in the modal. It should call the `confirmRejectionEvent` with `ARCHIVE_DECLARATION` action. In `submissionMiddleware`, handle the `ARCHIVE_DECLARATION` case similarly to  `REJECT_DECLARATION`. - 1
- [x] Modify the `markEventAsArchived` mutation to take in optional reason and comment as input similar to the `markEventAsVoided`. If reason and comment are given then add them to the task otherwise we need to make sure that comment and reason from previous task doesn't propagate. - 1
- [x] In `ActionDetailsModal`, for reject reason show the `Marked as duplicate` pill instead of the table. Check if the reason contains the value `duplicate`(Add a comment there that we are checking the value contains `duplicate` for legacy reasons. For Archived action make sure the comment and pill shows up if given. - 1

New tasks for mark as duplicate event
- [x] Create a new mutation in gateway `markEventAsDuplicate` which would take id, comment, reason and an optional duplicateTrackingId as input. If the duplicateTrackingId is provided, then add an extension to the task with that tracking id as value. Use `getTaskEntry` and `taskBundleWithExtension` functions - 2
- [x] We need to do the duplicateTrackingId part for the archive mutation as well. If provided in the input add an extension. - 1
- [x] Forward the taskBundle to workflow, add a new event in workflow similar which would detect the event and call actionEventHandler similar to #4120. Take a look at this commit for help: https://github.com/opencrvs/opencrvs-core/commit/b678719d74344d65d8e98a2244ed48469b6e2f18 - 2
- [x] In submissionMiddleware.ts for archive event, if the reason is equal to duplicate then we need to call the markAsDuplicate mutation, wait for it succeed before proceeding with the normal archive mutation. - 2
- [x] The marked as a duplicate action should show up in record audit and clicking it should open up modal according to design - 1
- [x] Display `Duplicate of table` on top of comment.  [Reference](https://www.figma.com/file/O6PevbBv0lApmXWYotR8sf/OpenCRVS-Design-Specifications-v2?node-id=11081%3A99867&t=CIGEYnNdSKaxsIIZ-4) - 1

**Comment by naftis** [comment]
@jpye-finch we need to discuss the difference the record audit is represented between THIS case and the potential duplicates.

**Comment by naftis** [comment]
Unblocked as jpye-finch added a design how the record audit should differ between this case and the other case: https://github.com/opencrvs/opencrvs-core/issues/4470 See the last acceptance criteria

**Comment by SyedaAfrida** [comment]
@anny320 this ticket is fixed and can be closed
https://www.loom.com/share/1cd75e7aa5f64c5db6e1b8df1a9dd09f

**Comment by anny320** [comment]
Completed
